File: application/src/main/java/com/github/dkurata38/open_web_library/application/book/BookRepository.java
package com.github.dkurata38.open_web_library.application.book;

import com.github.dkurata38.open_web_library.domain.book.Book;
import com.github.dkurata38.open_web_library.domain.book.BookId;
import org.springframework.data.repository.CrudRepository;

public interface BookRepository extends CrudRepository<Book, BookId> {
}

File: application/src/main/kotlin/com/github/dkurata38/open_web_library/application/book/BookRepository.kt
package com.github.dkurata38.open_web_library.application.book

import com.github.dkurata38.open_web_library.domain.book.Book
import com.github.dkurata38.open_web_library.domain.book.ISBN

interface BookRepository {
	fun findByISBN(isbn: ISBN): Book?

	fun save(book: Book)
}
